freedomofpress/securedrop
# Qualify Debian packages with target distribution

## Status

Ready for review

## Description of Changes

Includes the Ubuntu distribution in our .deb package names. Tries to normalize what we call the distribution/release/platform in all the packaging and deployment tooling.  Reduces the number of places `update_version.sh` has to update.

Partially addresses https://github.com/freedomofpress/securedrop-dev-packages-lfs/issues/85 by allowing `reprepro` to add all the files for both Ubuntu releases.

## Testing

- `git checkout -b qualify-debs origin/qualify-debs`
- `make build-debs`
- `make build-debs-focal`
- verify that all packages have the distribution in their names
- clone the `securedrop-dev-packages-lfs` repo
- in the root of the `securedrop-dev-packages-lfs` repo, run `sed -i 's/SignWith.*//' repo/conf/distributions` so `reprepro` doesn't complain that it can't sign files. 
- `reprepro --basedir ~/src/fpf/securedrop-dev-packages-lfs/repo --outdir /tmp/reprepro/ --dbdir /tmp/repreprodb includedeb xenial ~/src/fpf/securedrop/build/xenial/*.deb`
- `reprepro --basedir ~/src/fpf/securedrop-dev-packages-lfs/repo --outdir /tmp/reprepro/ --dbdir /tmp/repreprodb includedeb focal ~/src/fpf/securedrop/build/focal/*.deb`

There should be no errors from the second `reprepro` run about [files with different hashes](https://github.com/freedomofpress/securedrop-dev-packages-lfs/issues/85#issuecomment-765677012). The repo should contain the full set of files for each Ubuntu distribution.

- Try a staging environment for each distribution, e.g. `make staging` and `make staging-focal`, or run the configuration tests.

## Deployment

This should just affect package naming and metadata.

## Checklist

### If you made changes to the server application code:


### If you made changes to the system configuration:


### If you made non-trivial code changes:


Choose one of the following:
